src/rules/selector-pseudo-element-no-unknown/index.js

```diff
@@ -0,0 +1,63 @@
+import { isString } from "lodash"
+import selectorParser from "postcss-selector-parser"
+import { vendor } from "postcss"
+import {
+  isKnownPseudoElement,
+  report,
+  ruleMessages,
+  validateOptions,
+} from "../../utils"
+
+export const ruleName = "selector-pseudo-element-no-unknown"
+
+export const messages = ruleMessages(ruleName, {
+  rejected: (u) => `Unexpected unknown pseudo-element selector "${u}"`,
+})
+
+// level 1 and 2 pseudo elements
+const pseudoElements = new Set([
```

Comment:
Can we pull this out into a `isKnownLevel2PseudoElement` util that `isKnownPseudoElement` consumes?

We might as well get this rule done and dusted rather than doing that in another PR.